crates/cdk-sql-common/src/mint/mod.rs

```diff
@@ -789,7 +772,41 @@ where
             Amount::ZERO
         };
 
-        // Calculate new amount_paid with overflow check
+        // Step 2: Try to insert payment_id - this will fail fast if it's a duplicate
+        // The UNIQUE constraint on payment_id will prevent duplicate insertions atomically
+        let insert_result = query(
+            r#"
+            INSERT INTO mint_quote_payments
+            (quote_id, payment_id, amount, timestamp)
+            VALUES (:quote_id, :payment_id, :amount, :timestamp)
+            "#,
+        )?
+        .bind("quote_id", quote_id.to_string())
+        .bind("payment_id", payment_id.clone())
+        .bind("amount", amount_paid.to_i64())
+        .bind("timestamp", unix_time() as i64)
+        .execute(&self.inner)
+        .await;
+
+        // Check if insert failed due to duplicate payment_id
+        match insert_result {
+            Ok(_) => {
+                // Insert succeeded - now safe to update the amount
+            }
+            Err(err) => {
+                // Check if error is due to UNIQUE constraint violation
+                let err_msg = err.to_string().to_lowercase();
```

Comment:
Can this be done for each driver? As we do with [sqlite](https://github.com/cashubtc/cdk/blob/d9e001bee69674cdd5905b39750ebb947640c2b4/crates/cdk-sqlite/src/async_sqlite.rs#L62).